pilot/pkg/networking/core/v1alpha3/name_table.go

```diff
@@ -64,7 +64,7 @@ func (configgen *ConfigGeneratorImpl) BuildNameTable(node *model.Proxy, push *mo
 		// as two different TCP services are consuming the
 		// same wildcard passthrough TCP listener 0.0.0.0:3306.
 		//
-		if svc.Hostname.IsWildCarded() {
+		if svc.Hostname.IsWildCarded() && !hasHttpPorts(svc) {
```

Comment:
Why is it special for http protocol?

The Serviceentry.Addresses could be CIDR?

Reply:
It is because for TCP we can not determine fully - there can be conflicts. See the comments above.

We not process CIDRs while building name table

Reply:
TBH I do not understand the comment. Say I have `host: *.mysql.aws.com, port 3306, address: 1.2.3.4`. We can then match on 1.2.3.4:3306 and un-ambiguously match the correct cluster.

The only issue is if they have ORIGINAL_DST *and* a dummy IP. But that issue still happens with TCP and HTTP, since its at the cluster level.

Am I misunderstanding something?

Regardless - the comment probably needs some updates: `we cannot take services with wildcards in the address field.`

Reply:
Thinking about it more - I do not think we need not  have restriction on wildcards at all at DNS level.